crates/loro-internal/src/diff_calc.rs

```diff
@@ -434,12 +434,26 @@ impl DiffCalculatorTrait for MapDiffCalculator {
     }
 }
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+#[derive(Debug, Clone, PartialEq, Eq)]
 struct CompactMapValue {
     lamport: Lamport,
     peer: PeerID,
     counter: Counter,
-    value: Option<u32>,
+    value: Option<LoroValue>,
+}
+
+impl Ord for CompactMapValue {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.lamport
+            .cmp(&other.lamport)
+            .then(self.peer.cmp(&other.peer))
+    }
+}
+
+impl PartialOrd for CompactMapValue {
```

Comment:
I need also impl these traits manually for the tree.

Reply:
I think there is no easier way to bypass this